src/mg/PAMI/Client/Impl/ClientImpl.php

```diff
@@ -185,9 +189,7 @@ public function open()
 	    @stream_set_blocking($this->_socket, 0);
 	    $this->_currentProcessingMessage = '';
 	    //register_tick_function(array($this, 'process'));
-	    if ($this->_logger->isDebugEnabled()) {
```

Comment:
> 5. Finally, these CPU cycles can be won back (and then some) by upgrading to the latest PHP versions. These usually have so many performance improvements that the cost of having these calls are dwarfed by it. Since we are targeting people who run Asterisk, we have excluded almost by definition the users on shared hosting. Therefore, the remaining users will usually be able to do such an upgrade.

FYI: You might be shoked, but some FreePBX users seem to stuck in a timewarp of sorts, for example people using Elastix 2.5 (Centos 5.5 / Asterisk-1.8 / php 5.1 / non patched bash / nasty openssl). Upgrading php could be very problematic for them. 

I think they are nuts, running such old version of "everything", but they are out there. This is just to counter your assumption that not running in a hosted environment means up to date / updatable. This does not mean i would like to hold you back from using/implementing PSR3 and making people move forward.